libraries/shared/fetcher/geth_rpc_storage_fetcher_test.go

```diff
@@ -56,15 +59,13 @@ func (streamer *MockStoragediffStreamer) SetPayloads(payloads []statediff.Payloa
 
 var _ = Describe("Geth RPC Storage Fetcher", func() {
```

Comment:
Wanna see more tests relating to error handling, maybe including (some might be irrelevant, you know best :) )
- Unordered diffs
- Missed diffs (can we notice and re-fetch?)
- Trying to get data outside pruning limit
- Attached node isn't an archive node when we expect it to

Don't wanna be that guy, but it's a bit risky to add such large changes quite late, so let's not skimp on testing even though there is a bit of a hurry :blush:

Reply:
Found a lot more tests further down, but maybe some of these points are still useful :ok_hand:

Reply:
We decided we are not handling/retaining ordering of diffs at this point. We need to first anchor our diffs to a header so that we can check whether or not a diff has been received at a certain block if we want to handle missed diffs. I am getting conflicting feedback on whether or not it is safe to assume the node has the endpoint or not but I can add additional guards for that.